crates/script/src/receipts.rs

```diff
@@ -30,13 +30,17 @@ impl From<AnyTransactionReceipt> for TxStatus {
 /// the deploy sequence's pending vector
 pub async fn wait_for_pending(
     provider: Arc<RetryProvider>,
+    sequence_idx: usize,
     deployment_sequence: &mut ScriptSequence,
+    progress: &ScriptProgress,
```

Comment:
now that this requires &ScriptProgress, could we make this a function of ScriptProgress instead?

some with clear_pendings